bionic/core/flow_execution.py

```diff
@@ -110,12 +112,13 @@ def _process_active_entry(self, entry):
 
         assert entry.stage == EntryStage.ACTIVE
 
-        # In theory we could have an entry that doesn't require priming, but in practice
-        # we always require it when we create one.
-        assert entry.required_level >= EntryLevel.PRIMED
+        # In theory we could have an entry that doesn't require initializing, but in
+        # practice we always require at least that much when we create one.
+        assert entry.required_level >= EntryLevel.INITIALIZED
 
         # First, if we've already met our requirements, we can stop immediately without
         # needing to process any dependencies.
+        self._unblock_entries_with_met_requirements(entry)
```

Comment:
Yes, this makes me uncomfortable too -- I think because of the implicit requirement that we do this every time an entry's level could have changed. Of course, this requirement is specific to `_process_active_entry`, which already has an idiosyncratic structure.

The `_attempt_to_mark_complete` idea is interesting -- I'll try it out. I wonder if we can combine the "check if complete", "check if blocked", and "check for met requirements" into some kind of `_check_if_processing_should_continue` method.

Reply:
I pushed a commit that refactors this (among others). Let me know what you think.